Doc/tools/check-warnings.py

```diff
@@ -24,28 +26,150 @@
     "venv",
 }
 
-PATTERN = re.compile(r"(?P<file>[^:]+):(?P<line>\d+): WARNING: (?P<msg>.+)")
+# Regex pattern to match the parts of a Sphinx warning
+WARNING_PATTERN = re.compile(
+    r"(?P<file>([A-Za-z]:[\\/])?[^:]+):(?P<line>\d+): WARNING: (?P<msg>.+)"
+)
 
+# Regex pattern to match the line numbers in a Git unified diff
+DIFF_PATTERN = re.compile(
+    r"^@@ -(?P<linea>\d+),(?P<removed>\d+) \+(?P<lineb>\d+),(?P<added>\d+) @@",
+    flags=re.MULTILINE,
+)
 
-def check_and_annotate(warnings: list[str], files_to_check: str) -> None:
+
+def get_diff_files(ref_a: str, ref_b: str, filter_mode: str = "") -> set[Path]:
+    """List the files changed between two Gif refs, filtered by change type."""
+    added_files_result = subprocess.run(
+        [
+            "git",
+            "diff",
+            f"--diff-filter={filter_mode}",
+            "--name-only",
+            f"{ref_a}...{ref_b}",
+            "--",
+        ],
+        stdout=subprocess.PIPE,
+        check=True,
+        text=True,
+    )
+
+    added_files = added_files_result.stdout.strip().split("\n")
+    return {Path(file.strip()) for file in added_files if file.strip()}
```

Comment:
Does git return absolute paths?
```suggestion
    return {Path(file.strip()).resolve() for file in added_files if file.strip()}
```

Reply:
No, but that's intentional, because (like the previous version) we're only using `in` checks on them, and it effectively avoids the need for a couple extra steps parsing and normalizing both paths and possible edge cases with the same, as well as makes the `sphinx-warnings.txt` portable across repos/worktrees and locally/CI, both of which I took advantage of when developing, testing and trying this out.